src/runner.ts

```diff
@@ -0,0 +1,95 @@
+import fs from 'fs/promises';
+import path from 'path';
+import { globSync } from 'fast-glob';
+import { Config } from './types/config.t';
+import { Processor } from './processor';
+import { ASTKind, ASTReader, SourceUnit, compileSol } from 'solc-typed-ast';
+
+export class Runner {
+  config: Config;
+
+  constructor(config: Config) {
+    this.config = config;
+  }
+
+  async run() {
+    let sourceUnits = await this.getProjectCompiledSources(this.config.root, this.config.contracts, this.ignoredPaths());
+    if (!sourceUnits.length) return console.error('No solidity files found in the specified directory');
+
+    const processor = new Processor(this.config);
+    const warnings = processor.processSources(sourceUnits);
+
+    warnings.forEach(({ location, messages }) => {
+      console.warn(location);
+      messages.forEach((message) => {
+        console.warn(`  ${message}`);
+      });
+      console.warn();
+    });
+  }
+
+  ignoredPaths(): string[] {
```

Comment:
I'm not a fan of this. This function doesn't seem like it should be on the runner, instead sounds more of a Config class

Reply:
The runner ended up being some sort of utils class, and the run function will also be very hard to test. Imo, the run function should still live in main.ts, and the other functions should stay as utility functions